softioc/device.py

```diff
@@ -351,11 +351,29 @@ class ao(ProcessDeviceSupportOut):
     _ctype_ = c_double
     _dbf_type_ = fields.DBF_DOUBLE
 
+def to_epics_str_array(value):
+    """Convert the given array of Python strings to an array of EPICS
+    nul-terminated strings"""
+    result = numpy.empty(len(value), 'S40')
+
+    for n, s in enumerate(value):
+        if isinstance(s, str):
+            val = EpicsString._ctype_()
+            val.value = s.encode() + b'\0'
+            result[n] = val.value
+        else:
+            result[n] = s
```

Comment:
Hmm.  Does this do the right thing when `s` is a strange value?

Reply:
I added some tests for mixed lists in 5b59ed7f86a126842467247dfb004ecf21aed561 - i.e. mixing strings and integers in a single list.